apps/settings/js/panels/call/call_settings.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Because there are some APIs that are working like mutex, we have to 
+ * wrap them inside a special module and provide a simple interface for
+ * users to use.
+ *
+ * And also, we can use CallSettings.observe('isLocked', ...) to know
+ * what's the current status and do related UI stuffs (like blocking ...)
+ */
+define(function(require) {
+  'use strict';
+
+  var SettingsCache = require('modules/settings_cache');
+  var TaskScheduler = require('modules/task_scheduler');
+  var CallForwarding = require('panels/call/call_forwarding');
+
+  var CallSettings = function() {
+    this._cardIndex = 0;
+    this._conns = window.navigator.mozMobileConnections;
+    this._conn = this._conns[this._cardIndex];
+    this._taskScheduler = TaskScheduler();
```

Comment:
We should have the same number of task schedulers as the number of mobile connection instances because they can work in parallel.